datasets/us/fda_restricted/crawler.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict
+
+from zavod import Context, helpers as h
+import re
+
+NAME_PATTERN = re.compile(
+    r"""
+    ^\s*([A-Z][a-zA-Z' -]+)          # Last name(s)
+    (?:,\s*(Jr\.?|Sr\.?|II|III|IV))? # Optional suffix
+    (?:,\s*(MD|PhD|DO|DVM))?         # Optional pre-first-name credential
+    ,\s*([A-Z][a-zA-Z]+)             # First given name
+    (?:\s+([A-Z](?:\.)?|[A-Z][a-zA-Z]+(?:\s+[A-Z](?:\.)?|[A-Z][a-zA-Z]+)*))? # Additional given names/initials
+    (?:,\s*(.*))?                    # Remaining credentials (best effort)
+    \s*$""",
+    re.VERBOSE,
+)
+
+
+def crawl_item(row: Dict[str, str], context: Context):
+
+    raw_name = row.pop("name")
+    address = h.make_address(
+        context, city=row.pop("city"), state=row.pop("state"), country_code="us"
+    )
+
+    entity = context.make("Person")
+    entity.id = context.make_id(raw_name)
+
+    match = NAME_PATTERN.match(raw_name)
+    if not match:
+        context.log.warning(f"Unable to parse {raw_name}")
+        entity.add("name", raw_name)
+    else:
+        last, suffix, precred, first, middle, postcreds = match.groups()
+        h.apply_name(
+            entity,
+            last_name=last,
+            suffix=suffix,
+            first_name=first,
+            middle_name=middle,
+            prefix=postcreds,
+        )
+
+    entity.add("country", "us")
+    entity.add("address", address)
+    entity.add("sector", row.pop("center"))
+
+    sanction = h.make_sanction(context, entity)
+    h.apply_date(sanction, "startDate", row.pop("date_of_status"))
+    sanction.add("status", row.pop("status"))
+
+    # In those cases the sanction is not active anymore
+    target = sanction.get("status") not in [
+        "Restrictions Removed",
+        "Not Disqualified",
+        "Reinstated pursuant to Agreement",
+        "Not Disqualified – Adequate Assurances 1",
+        "Restrictions Removed (reinstated by correspondence)",
+    ]
+
+    if target:
+        entity.add("topics", "debarment")
+
+    context.emit(entity, target=target)
+    context.emit(sanction)
+
+    # NIDPOE = Notice of Initiation of Disqualification Proceedings and Opportunity to Explain
+    # NOOH = Notice of Opportunity for Hearing
+    context.audit_data(
+        row,
+        ignore=[
+            "date_nidpoe_issued",
+            "date_nooh_issued",
+            "link_to_nidpoe_letter",
+            "link_to_nooh_letter",
+            "date_of_presiding_officer_report",
+            "link_to_presiding_officer_report",
+            "date_of_commissioner_s_decision",
+            "link_to_commissioner_s_decision",
```

Comment:
let's put all the links into sanction:sourceUrl - it looks like the PDF links don't wok but the text ones do? you can get them from the html of the original `row` returned by the table parser.